setup.py

```diff
@@ -25,17 +25,19 @@
     subprocess.call([sys.executable, "-m", "pip", "install", "Cython>=0.27.1"])
     from Cython.Build import cythonize
 
+try:
+    from setuptools_rust import Binding, RustExtension
+except ImportError:
+    import subprocess
+
+    subprocess.call([sys.executable, "-m", "pip", "install", "setuptools-rust"])
+    from setuptools_rust import Binding, RustExtension
+
```

Comment:
Yeah we should just move to just relying on the pyproject.toml here, I just copied it because we were doing it for cython. You need pip >=19.0 for it to support pep517 and use the pyproject.toml, which at one time (especially when we first added cython to the repo) was an issue . But now I think it's probably been long enough that we can move away from doing this.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7658/commits/7af1c4331d9f26c27b9ad5f101ff5a8c6bc02bac